tests/order_modifiers_integration/Checkout/Gateway/PayPal/Coupons_Test.php

```diff
@@ -0,0 +1,166 @@
+
+		// Create the order using the REST API, because that's how it will work on a live site.
+		[ $path, $query_args, $args ] = $this->create_order_via_rest();
+
+		// Verify the data from the request.
+		Assert::assertEquals( '/v2/checkout/orders', $path );
+		Assert::assertEmpty( $query_args );
+		Assert::assertTrue( isset( $args['body']['purchase_units'][0] ) );
+
+		$purchase_unit = $args['body']['purchase_units'][0];
+		Assert::assertCount( 1, $purchase_unit['items'], 'There should be a single item in the unit data' );
+		Assert::assertEquals( 2, $purchase_unit['items'][0]['quantity'], 'The quantity should be 2' );
+		Assert::assertArrayHasKey( 'breakdown', $purchase_unit['amount'], 'The amount should have a breakdown of items.' );
+		Assert::assertArrayHasKey( 'item_total', $purchase_unit['amount']['breakdown'], 'The breakdown should have an item total' );
+		Assert::assertArrayNotHasKey( 'discount', $purchase_unit['amount']['breakdown'], 'The breakdown should not have a discount' );
+
+		// The item total should be 11.28 * 2 = 22.56 (as a string).
+		Assert::assertEquals( '22.56', $purchase_unit['amount']['breakdown']['item_total']['value'] );
+	}
+
+	/**
+	 * @test
+	 */
+	public function it_should_modify_paypal_unit_data() {
+		$this->make_controller()->register();
+
+		/**
+		 * @var int    $ticket_id
+		 * @var Coupon $coupon
+		 */
+		[ $ticket_id, $coupon ] = $this->create_the_things();
+
+		// Add the ticket and coupon to the cart.
+		/** @var Commerce_Cart $cart */
+		$cart = tribe( Commerce_Cart::class );
+		$cart->add_ticket( $ticket_id, 2 );
+		$coupon->add_to_cart( $cart->get_repository() );
+
+		// Create the order using the REST API, because that's how it will work on a live site.
+		[ $path, $query_args, $args ] = $this->create_order_via_rest();
+
+		// Verify the data from the request.
+		Assert::assertEquals( '/v2/checkout/orders', $path );
+		Assert::assertEmpty( $query_args );
+		Assert::assertTrue( isset( $args['body']['purchase_units'][0] ) );
+
+		$purchase_unit = $args['body']['purchase_units'][0];
+		Assert::assertCount( 1, $purchase_unit['items'], 'There should be a single item in the unit data' );
+		Assert::assertEquals( 2, $purchase_unit['items'][0]['quantity'], 'The quantity should be 2' );
+		Assert::assertArrayHasKey( 'breakdown', $purchase_unit['amount'], 'The amount should have a breakdown of items.' );
+		Assert::assertArrayHasKey( 'item_total', $purchase_unit['amount']['breakdown'], 'The breakdown should have an item total' );
+		Assert::assertArrayHasKey( 'discount', $purchase_unit['amount']['breakdown'], 'The breakdown should have a discount' );
+
+		// The item total should be 11.28 * 2 = 22.56 (as a string).
+		Assert::assertEquals( '22.56', $purchase_unit['amount']['breakdown']['item_total']['value'] );
+
+		// The discount should be 17.3% * 2 * 11.28 = 3.90 (as a string).
+		Assert::assertEquals( '3.90', $purchase_unit['amount']['breakdown']['discount']['value'] );
```

Comment:
I think adding one more Assert to make sure that the item_total value + discount value = the complete total would also be a good check.